### PR TITLE
SceGxm: add missing returns before RET_ERROR

### PR DESCRIPTION
**Pull request by scribam**
And remove some invalid checks in `sceGxmTextureGet*` functions (verified by RE)

### CODE REVIEW

File: vita3k/modules/SceGxm/SceGxm.cpp [hunk 1]
```diff
@@ -755,7 +755,7 @@ EXPORT(void, sceGxmFinish, SceGxmContext *context) {
     assert(context);
 
     if (!context)
-        RET_ERROR(SCE_GXM_ERROR_INVALID_POINTER);
+        return;
```
Comment (Zangetsu38) [inline vita3k/modules/SceGxm/SceGxm.cpp:758]: return nothing normal ?

Comment (scribam) [reply vita3k/modules/SceGxm/SceGxm.cpp:758]: The return type of the function is `void` so yes, it is normal

Comment (Zangetsu38) [reply vita3k/modules/SceGxm/SceGxm.cpp:758]: oh, i see lol

**Review by sunho — APPROVED** [review]
Thanks for doing this

**Comment by sunho** [comment]
IIRC returning made regressions in many games (e.g. new game, p4g) Even if this is from RE, we better keep track of how these changes affect compatibility.

**Comment by scribam** [comment]
I tried a new game in P4G and haven't found any issues so far. Only the changes in `sceGxmTextureGet*` functions are from RE, the other ones are to "fix" the fact `RET_ERROR` only doesn't not return if you don't add a return before.
I plan to push more changes about error checking after this PR.

**Pull request merged**